### PR TITLE
allow a scene to have both transmission and reflection

### PR DESCRIPTION
**Pull request by pixelflinger**
what's not allowed at this point is an object which has both screen
space refraction and reflection.

this is accomplished by just always disabling screen-space reflections
for the translucent step of the refraction pass
Screen Shot 2022-01-18 at 2 44 41 PM
.

### CODE REVIEW

**Review by prideout — APPROVED** [review]

**Review by bejado — APPROVED** [review]
So easy!

**Review by romainguy — APPROVED** [review]

File: filament/src/Renderer.cpp [hunk 1]
```diff
@@ -972,32 +975,23 @@ FrameGraphId<FrameGraphTexture> FRenderer::colorPass(FrameGraph& fg, const char*
...
+                            getClipSpaceToTextureSpaceMatrix() *
+                            cameraInfo.projection;
+                    reprojection =
+                            getClipSpaceToTextureSpaceMatrix() *
+                            historyProjection *
+                            inverse(cameraInfo.view * cameraInfo.worldOrigin);
                 }
+                TextureHandle ssrHandle = data.ssr ?
+                        resources.getTexture(data.ssr) : ppm.getOneTexture();
+                view.prepareSSR(ssrHandle,config.refractionLodOffset,
```
Comment (bejado) [inline filament/src/Renderer.cpp:992]:
```suggestion
                view.prepareSSR(ssrHandle, config.refractionLodOffset,
```

File: filament/src/Renderer.cpp [hunk 2]
```diff
@@ -972,32 +975,23 @@ FrameGraphId<FrameGraphTexture> FRenderer::colorPass(FrameGraph& fg, const char*
                         resources.getTexture(data.structure) : ppm.getOneTexture());
 
                 // set screen-space reflections or screen-space refractions
```
Comment (bejado) [inline filament/src/Renderer.cpp:977]:
```suggestion
                // set screen-space reflections and screen-space refractions
```

**Review by bejado — CHANGES_REQUESTED** [review]
Can you update [this comment](https://github.com/google/filament/blob/4b572cf0657ce49b7e5aafd8420072655f6e71e7/libs/filamat/src/shaders/ShaderGenerator.cpp#L311-L314) in ShaderGenerator.cpp? Other than that, LGTM!

**Pull request merged**